server/src/main/java/org/elasticsearch/index/mapper/AbstractGeometryFieldMapper.java

```diff
@@ -184,6 +177,27 @@ public Builder ignoreZValue(final boolean ignoreZValue) {
             return this;
         }
 
+        public T docValues(boolean hasDocValues) {
+            // TODO(talevy): see how to best make this pluggable with the DataHandler work
+            if (hasDocValues) {
+                throw new ElasticsearchParseException("field [" + name + "] of type [" + fieldType().typeName()
+                    + "] does not support doc_values");
+            }
+            return super.docValues(hasDocValues);
+        }
+
+        protected Explicit<Boolean> docValues() {
+            // TODO(talevy): see how to best make this pluggable with the DataHandler work
+            // although these values can be true, an ElasticsearchParseException
+            // prevents this Explicit(true,true) path to ever occur in practice
```

Comment:
One possibility would be to only enable doc values by default if there is a registered DataHandler.